### PR TITLE
Revamp reporting bugs docs

### PR DESCRIPTION
**Pull request by vitorgalvao**
The eternal struggle to make the reporting bugs docs read and understood by more users. This is part of a process of changing the error messages on the core to link directly to these fixes, instead of general bug reports.

### CODE REVIEW

**Comment by vitorgalvao** [comment]
First suggestion to the core, incorporating this: https://github.com/Homebrew/brew/pull/5595

**Review by claui — CHANGES_REQUESTED** [review]
While we’re touching the docs, maybe iron out a few small bumps in the copy.

File: doc/reporting_bugs/checksum_does_not_match_error.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+First, check if the problem was with your download. Delete the downloaded file (its location will be pointed out right under the `Actual` shasum line) and try again.
+
+If the problem persists, the cask must be outdated. It’ll likely need a new version, but it’s possible the version has remained the same (happens occasionally when the vendor updates the app in place).
+
+1. Go to the vendor’s website (`brew cask home {{cask_name}}`).
+    * Alternatively, if it has an `appcast`, read that (`curl "$(brew cask _stanza appcast {{cask_name}})"`).
+2. Find out what the latest version is. It will likely be expressed in the URL used to download it.
+3. Take a look at the cask’s version (`brew cask _stanza version {{cask_name}}`) and verify it is indeed outdated. if it is:
```
Comment (claui) [inline doc/reporting_bugs/checksum_does_not_match_error.md]:
```suggestion
3. Take a look at the cask’s version (`brew cask _stanza version {{cask_name}}`) and verify it is indeed outdated. If it is:
```

**Review by claui — CHANGES_REQUESTED** [review]

**Review by claui — APPROVED** [review]
LGTM now 👍

**Comment by vitorgalvao** [comment]
Thank you for the fixes, @claui!

**Pull request merged**